app_flutter/lib/commit_box.dart

```diff
@@ -5,29 +5,108 @@
 import 'package:flutter/material.dart';
 
 /// Displays Git commit information.
-class CommitBox extends StatelessWidget {
+class CommitBox extends StatefulWidget {
   // TODO(chillers): convert to use commit model
-  const CommitBox({
-    Key key,
-    @required this.message,
-    @required this.avatarUrl,
-  }) : super(key: key);
+  const CommitBox(
+      {Key key,
+      @required this.message,
+      @required this.avatarUrl,
+      @required this.author})
+      : super(key: key);
 
   /// Commit message that summarizes the change made.
   final String message;
 
   /// Image URL to the avatar of the author of this commit.
   final String avatarUrl;
 
+  /// The person that authored this commit.
+  final String author;
+
+  @override
+  _CommitBoxState createState() => _CommitBoxState();
+}
+
+class _CommitBoxState extends State<CommitBox> {
+  OverlayEntry _commitOverlay;
+  final LayerLink _layerLink = LayerLink();
+
   @override
   Widget build(BuildContext context) {
-    // TODO(chillers): add overlay to view more information
-    return Container(
-      margin: const EdgeInsets.all(1.0),
-      child: Image.network(
-        avatarUrl,
-        height: 40,
+    return GestureDetector(
+      onTap: () {
+        this._commitOverlay = this._createCommitOverlay(widget);
+        Overlay.of(context).insert(this._commitOverlay);
+      },
+      child: Container(
+        margin: const EdgeInsets.all(1.0),
+        child: Image.network(
+          widget.avatarUrl,
+          height: 40,
+        ),
       ),
     );
   }
+
+  OverlayEntry _createCommitOverlay(CommitBox widget) {
+    RenderBox renderBox = context.findRenderObject();
+
+    return OverlayEntry(
+        builder: (context) => Stack(
+              children: <Widget>[
+                // This is the area a user can click (the rest of the screen) to close the overlay.
+                GestureDetector(
+                  onTap: () {
```

Comment:
Use tear-offs here:

```suggestion
                  onTap: _commitOverlay.remove
```

Reply:
Done, thanks for sharing the name of the syntax!